### ISSUE TITLE
Box: BoxProps incompatible with Box component

### ISSUE THREAD

**Issue by vlad-nica** [freenowtech/wave#190]
-   `@freenow/wave` version: 1.16.1
-  `typescript` version: 4.4.4

### Relevant code

```js
import { Box, BoxProps } from '@freenow/wave'

const SomeComponent = (props: BoxProps) => {
    return <Box {...props} />
}
```

### What was expected to happen?

The code above throws a typescript error:
```
No overload matches this call.
  Overload 1 of 2, '(props: Omit<Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "key" | keyof HTMLAttributes<...>> & { ...; } & BoxProps, "theme"> & Partial<...>, "theme"> & { ...; } & { ...; }): ReactElement<...>', gave the following error.
    Type '{ m?: ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>> | undefined; margin?: ResponsiveValue<string | number | symbol, Required<...>> | undefined; ... 78 more ...; textAlign?: ResponsiveValue<...> | undefined; }' is not assignable to type 'Omit<Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "key" | keyof HTMLAttributes<HTMLDivElement>> & { ...; } & BoxProps, "theme"> & Partial<...>, "theme">'.
      Types of property 'color' are incompatible.
        Type 'ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>> | undefined' is not assignable to type '(string & ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>>) | undefined'.
          Type 'null' is not assignable to type '(string & ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>>) | undefined'.
  Overload 2 of 2, '(props: StyledComponentPropsWithAs<"div", never, BoxProps, "theme", "div", "div">): ReactElement<StyledComponentPropsWithAs<"div", never, BoxProps, "theme", "div", "div">, string | JSXElementConstructor<...>>', gave the following error.
    Type '{ m?: ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>> | undefined; margin?: ResponsiveValue<string | number | symbol, Required<...>> | undefined; ... 78 more ...; textAlign?: ResponsiveValue<...> | undefined; }' is not assignable to type 'Omit<Omit<Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "key" | keyof HTMLAttributes<HTMLDivElement>> & { ...; } & BoxProps, "theme"> & Partial<...>, "theme">'.
      Types of property 'color' are incompatible.
        Type 'ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>> | undefined' is not assignable to type '(string & ResponsiveValue<string | number | symbol, Required<Theme<TLengthStyledSystem>>>) | undefined'.ts(2769)
```

**Comment by snapsnapturtle** [comment]
@vlad-nica can you let us know what typescript version you're using? can create a code sandbox example to reproduce this?

**Comment by vlad-nica** [comment]
Typescript version: 4.4.4 (added also in the original post).
CodeSandbox: https://codesandbox.io/s/wave-box-boxprop-z0wyv?file=/src/App.tsx

**Comment by jonotrujillo** [comment]
Fixed in #197
